File: C++/Medium/CompileError/path-with-maximum-gold_8.cpp
class Solution {
public:
    int getMaximumGold(vector<vector<int>>& grid) {
        
        int m = grid.size();
        int n = grid[0].size();
        vector<vector<bool>> visited(m, vector<bool>(n, false));
        int maxGold = 0;
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (grid[i][j] > 0) {
                    maxGold = max(maxGold, dfs(grid, i, j, visited));
                }
            }
        }
        return maxGold;
    }
};

File: Rust/Medium/WrongAnswer/path-with-maximum-gold_8.rs
impl Solution {
    pub fn get_maximum_gold(grid: Vec<Vec<i32>>) -> i32 {
        
        let mut result: i32 = 0;
        let mut visited: Vec<Vec<bool>> = vec![vec![false; grid[0].len()]; grid.len()];
        for i in 0..grid.len() {
            for j in 0..grid[0].len() {
                if grid[i][j] != 0 {
                    let mut visited: Vec<Vec<bool>> = vec![vec![false; grid[0].len()]; grid.len()];
                    let mut temp = 0;
                    Self::dfs(&grid, i, j, &mut visited, &mut temp);
                    result = result.max(temp);
                }
            }
        }
        result
    }
    fn dfs(grid: &Vec<Vec<i32>>, i: usize, j: usize, visited: &mut Vec<Vec<bool>>, temp: &mut i32) {
        if i >= grid.len() || j >= grid[0].len() || i < 0 || j < 0 || visited[i][j] || grid[i][j] == 0 {
            return;
        }
        visited[i][j] = true;
        *temp += grid[i][j];
        Self::dfs(grid, i + 1, j, visited, temp);
        Self::dfs(grid, i - 1, j, visited, temp);
        Self::dfs(grid, i, j + 1, visited, temp);
        Self::dfs(grid, i, j - 1, visited, temp);
        visited[i][j] = false;
        *temp -= grid[i][j];
    }
}